From 1e5a3e902eb9563ab80a03d03e40ec5a74899e37 Mon Sep 17 00:00:00 2001
From: Hyun Kwon <hyun.kwon@xilinx.com>
Date: Tue, 21 Jul 2020 09:13:05 -0700
Subject: [PATCH 1451/1852] v4l: xilinx: dprx: Fix incorrect comparisons

commit 2744714b8102000b1f0f77bf2c0a51777b8df008 from
https://github.com/Xilinx/linux-xlnx.git

By fixing the type of ret from u32 to int.

drivers/media/platform/xilinx/xilinx-dprxss.c: In function 'xdprxss_parse_of':
drivers/media/platform/xilinx/xilinx-dprxss.c:1028:10: warning: comparison of unsigned expression < 0 is always false [-Wtype-limits]
 1028 |  if (ret < 0) {
      |          ^
drivers/media/platform/xilinx/xilinx-dprxss.c:1059:10: warning: comparison of unsigned expression < 0 is always false [-Wtype-limits]
 1059 |  if (ret < 0) {
      |          ^
drivers/media/platform/xilinx/xilinx-dprxss.c:1073:10: warning: comparison of unsigned expression < 0 is always false [-Wtype-limits]
 1073 |  if (ret < 0) {
      |          ^
drivers/media/platform/xilinx/xilinx-dprxss.c:1084:10: warning: comparison of unsigned expression < 0 is always false [-Wtype-limits]
 1084 |  if (ret < 0) {

Signed-off-by: Hyun Kwon <hyun.kwon@xilinx.com>
Reviewed-by: Rajesh Gugulothu <gugulothu.rajesh@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-dprxss.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/media/platform/xilinx/xilinx-dprxss.c b/drivers/media/platform/xilinx/xilinx-dprxss.c
index 1e8cb2c8a31c..77aa00c20a6c 100644
--- a/drivers/media/platform/xilinx/xilinx-dprxss.c
+++ b/drivers/media/platform/xilinx/xilinx-dprxss.c
@@ -1027,7 +1027,8 @@ static const struct v4l2_subdev_ops xdprxss_ops = {
 static int xdprxss_parse_of(struct xdprxss_state *xdprxss)
 {
 	struct device_node *node = xdprxss->dev->of_node;
-	u32 val, ret;
+	u32 val;
+	int ret;
 
 	ret = of_property_read_u32(node, "xlnx,bpc", &xdprxss->bpc);
 	if (ret < 0) {
-- 
2.31.1

